node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeFailer.java

```diff
@@ -155,35 +155,38 @@ protected double maintain() {
         return failingNodes;
     }
 
-    private List<FailingNode> findActiveFailingNodes() {
+    private Collection<FailingNode> findActiveFailingNodes() {
+        Set<FailingNode> failingNodes = new HashSet<>();
         NodeList activeNodes = nodeRepository().nodes().list(Node.State.active);
-        Instant graceTimeEnd = clock().instant().minus(downTimeLimit);
-        List<FailingNode> failingNodes = new ArrayList<>();
+
         for (Node node : activeNodes) {
-            if (node.history().hasEventBefore(History.Event.Type.down, graceTimeEnd) && ! applicationSuspended(node)) {
+            Instant graceTimeStart = clock().instant().minus(suspended(node) ? suspendedDownTimeLimit : downTimeLimit);
+            if (node.history().hasEventBefore(History.Event.Type.down, graceTimeStart) && !applicationSuspended(node)) {
                 // Allow a grace period after node re-activation
-                if ( ! node.history().hasEventAfter(History.Event.Type.activated, graceTimeEnd))
+                if (!node.history().hasEventAfter(History.Event.Type.activated, graceTimeStart))
                     failingNodes.add(new FailingNode(node, "Node has been down longer than " + downTimeLimit));
             }
-            else if (hostSuspended(node, activeNodes)) {
-                Node hostNode = node.parentHostname().flatMap(parent -> nodeRepository().nodes().node(parent)).orElse(node);
-                if (hostNode.type().isHost()) {
-                    List<String> failureReports = reasonsToFailParentHost(hostNode);
-                    if (failureReports.size() > 0) {
-                        if (hostNode.equals(node)) {
-                            failingNodes.add(new FailingNode(node, "Host has failure reports: " + failureReports));
-                        } else {
-                            failingNodes.add(new FailingNode(node, "Parent (" + hostNode + ") has failure reports: " + failureReports));
-                        }
+        }
+
+        for (Node node : activeNodes) {
+            if (allSuspended(node, activeNodes)) {
+                Node host = node.parentHostname().flatMap(parent -> nodeRepository().nodes().node(parent)).orElse(node);
```

Comment:
Consider reusing the node snapshot from the start of the method here and in `hasHardwareIssue()`.

Looks like this class reads nodes too eagerly in general.